kubemarine/core/group.py

```diff
@@ -399,13 +428,16 @@ def _make_result_or_fail(self, results: _HostToResult,
         return group_result
 
     @_handle_internal_logging
+    @_handle_dry_run
     def run(self, *args, **kwargs) -> Union[NodeGroupResult, int]:
```

Comment:
I suggest to not call NodeGroup.run/sudo/put for RW operations in dry run at all. The reason is it is difficult to simulate its behaviour. For example, you return empty results, but in fact it returns not empty results in real run. So this can lead to unpredictable changes in behaviour of Kubemarine. Also, put does not return result at all, run/sudo might return int in some cases, and so on...